ir_datasets/datasets/sara.py

```diff
@@ -0,0 +1,225 @@
+    tokenised = word_tokenize(email_text)
+    return len(tokenised)
+
+def number_of_recipients(filename):
+    addresses = []
+    contents = save_email_from_filename(filename)
+    contents = contents.split("\n")
+    for line in contents:
+        if line.startswith("To:"):
+            addresses.append(re.findall(r"[a-z0-9\.\-+_]+@[a-z0-9\.\-+_]+\.[a-z]+", line))
+    
+    return len([j for sub in addresses for j in sub])
+
+def get_sender_from_filename(filename):
+    addresses = []
+    contents = save_email_from_filename(filename)
+    contents = contents.split("\n")
+    for line in contents:
+        if line.startswith("From:"):
+            addresses.append(re.findall(r"[a-z0-9\.\-+_]+@[a-z0-9\.\-+_]+\.[a-z]+", line))
+    
+    return [j for sub in addresses for j in sub][0]
+
+def search_by_id(id):
+    all_emails = list_all_filenames()
+    for email in all_emails:
+        text = save_email_from_filename(email)
+        if id in text:
+            print(text)
+
+def sensitivity_stats(filenames):
+  non_sensitive_count = 0
+  sensitive_count = 0
+  
+  for f in filenames:
+    if f in non_sensitive_filenames:
+      non_sensitive_count += 1
+    if f in sensitive_filenames:
+      sensitive_count += 1
+
+  print(f"Non-Sensitive: {non_sensitive_count} | Sensitive: {sensitive_count} ")
+
+def get_docs_frame(directory):
+    # Generate dataframe for collection to be indexed
+    data = []
+    # Partition emails into sensitive and non-sensitive
+    sensitive_filenames = []
+    # Assume 'Purely personal' and 'Personal but in a professional context' are the sensitive categories
+    sensitive_filenames.append(filenames_from_cat(1,2, directory))
+    sensitive_filenames.append(filenames_from_cat(1,3, directory))
+
+    # Flatten the list
+    sensitive_filenames = [j for sub in sensitive_filenames for j in sub]
+    # Remove duplicates - 3 emails are counted in both categories
+    sensitive_filenames = list(dict.fromkeys(sensitive_filenames))
+
+    non_sensitive_filenames = []
+    for name in list_all_filenames(directory):
+        if name not in sensitive_filenames:
+            non_sensitive_filenames.append(name)
+
+    for filename in sensitive_filenames:
+        email = save_email_from_filename(filename, directory)
+        data.append([filename,email,1])
+
+    for filename in non_sensitive_filenames:
+        email = save_email_from_filename(filename, directory)
+        data.append([filename,email,0])
+
+    return data
+
+# A unique identifier for this dataset. This should match the file name (with "-" instead of "_")
+NAME = "sara"
+
+# What do the relevance levels in qrels mean?
+QREL_DEFS = {
+    2: 'highly relevant',
+    1: 'partially relevant',
+    0: 'not relevant',
+}
+
+# This message is shown to the user before downloads are started
+DUA = 'Please confirm that you agree to the data usage agreement at <https://some-url/>'
+
+class SaraDoc(NamedTuple):
+    doc_id: str
+    text: str
```

Comment:
Wondering about the use case -- would it be helpful to parse the documents into the fields (e.g., subject, body, headers, etc.)? Rather than a freeform text that contains the email-encoded body,